lib/tasks/seek_upgrades.rake

```diff
@@ -156,6 +157,21 @@ namespace :seek do
     puts " ... finished updating sample_resource_links of #{samples_updated.to_s} samples with data_file attributes"
   end
 
+  task(recognise_isa_json_compliant_items: [:environment]) do
+    puts '... searching for ISA compliant investigations'
+    investigations_updated = 0
+    disable_authorization_checks do
+      isa_json_compliant_studies = Study.all.select { |study| study.sample_types.any? }
+      isa_json_compliant_studies.each do |study|
+        inv = study.investigation
+        inv.update(is_isa_json_compliant: true)
+        inv.save
+        investigations_updated += 1
+      end
```

Comment:
to be more effecient and repeatable, you could limit the studies to only those linked to an investigation where is_isa_json_compliant is false. Something like `Study.join(:investigation).where('study.investigation.is_isa_json_compliant = ?', false).select ...`  (probably not quite right, but you get the idea).
Also, it looks like it is updating the same investigation multiple times if it has multiple studies.

You should use `inv.update_column(:is_isa_json_compliant, true)` to avoid triggering callbacks and and changing timestamps